src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -684,6 +713,213 @@ IECore::DataPtr loadSpline(
 
 } // namespace
 
+void ShaderNetworkAlgo::convertToOSLConventions( ShaderNetwork *network, int oslVersion )
+{
+	std::map< IECore::InternedString, std::pair< IECore::InternedString, size_t > > currentSplineArrayAdapters;
```

Comment:
Does this need to be declared all the way up here? It's so far from the point of usage, with no explanation as to what the types mean, so it makes it pretty hard to understand.

Reply:
I'm not a huge fan of reallocating the map for every shader as a way of documenting its existence, but happy to add a comment where it's defined, or where it's used, or both.  And maybe some of the other changes you've requested will reduce the distance between the beginning of the loop and where it's used.